matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/internal/DefaultMatchingNotificationResource.java

```diff
@@ -343,6 +348,23 @@ private Response getMatchesResponse(@Nullable final double score, @Nullable fina
             matchesJson = buildMatchesJSONArray(matches);
             matchesJson.put(REQ_NO, reqNo);
         }
+
+        JSONObject params = new JSONObject();
+        params.put("minScore", score);
+        params.put("minPhenScore", phenScore);
+        params.put("minGenScore", genScore);
+
+        if (fromDate != null) {
+            params.put("fromDate", DATE_TIME_SDF.format(fromDate));
+        }
+
+        if (toDate != null) {
+            // need to add 1 day without the last millisecond in ms
+            // to let user know that we include all matches found that day
+            Timestamp timestampTo = new Timestamp(toDate.getTime() + ONE_DAY_IN_MILLISECONDS);
```

Comment:
we should update `toDate` only once, before calling `this.matchStorageManager.loadMatches()`, and then put here the value we pass there. Otherwise it may get out of sync

Reply:
fair